app/models/metric/targets.rb

```diff
@@ -58,16 +56,46 @@ def self.capture_container_targets(zone, _options)
     targets
   end
 
-  def self.capture_vm_targets(targets, parent_class, options)
-    return Vm.none if options[:exclude_vms]
-    enabled_parents = targets.select do |t|
-      t.kind_of?(parent_class) &&
-        t.kind_of?(Metric::CiMixin) &&
-        t.perf_capture_enabled? &&
-        t.respond_to?(:vms)
+  def self.preload_infra_targets_data(zone, options)
+    # Preload all of the objects we are going to be inspecting.
+    includes = {:ext_management_systems => {:hosts => {:ems_cluster => :tags, :tags => {}}}}
+    includes[:ext_management_systems][:hosts][:storages] = :tags unless options[:exclude_storages]
+    includes[:ext_management_systems][:hosts][:vms] = {} unless options[:exclude_vms]
+
+    MiqPreloader.preload(zone, includes)
+
+    # populate ems (with tags / clusters)
+    zone.ext_management_systems.each do |ems|
+      ems.hosts.each do |host|
+        host.ems_cluster.association(:ext_management_system).target = ems if host.ems_cluster_id
+        unless options[:exclude_vms]
```

Comment:
Sorry @kbrock, you're right. I didn't see 92. I'm ok with keeping as is for now.  A simple guard clause won't work here so it's probably better the way it is until/if this is broken up into methods.